src/promptflow/promptflow/executor/_process_manager.py

```diff
@@ -0,0 +1,219 @@
+import multiprocessing
+import queue
+import signal
+from functools import partial
+import psutil
+from promptflow._core.operation_context import OperationContext
+from promptflow._utils.logger_utils import LogContext, bulk_logger
+from promptflow.executor.flow_executor import FlowExecutor
+
+
+class ProcessManager:
```

Comment:
It would be better to add more document string to explain the usage of the classes.